android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -0,0 +1,205 @@
+        Log.d(TAG, "onTaskRemoved");
+        stopSelf();
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mLocalBinder;
+    }
+
+    // The callbacks for the ble advertisement events
+    private final AdvertiseCallback mAdvertiseCallback = new AdvertiseCallback() {
+
+        // Fires when the URL is successfully being advertised
+        @Override
+        public void onStartSuccess(AdvertiseSettings advertiseSettings) {
+            Log.d(TAG, "URL is broadcasting");
+            createNotification();
+            Toast.makeText(getApplicationContext(), getString(R.string.broadcasting),
+                Toast.LENGTH_LONG).show();
+        }
+
+        // Fires when the URL could not be advertised
+        @Override
+        public void onStartFailure(int result) {
+            Log.d(TAG, "onStartFailure" + result);
+        }
+    };
+
+
+
+    /////////////////////////////////
+    // utilities
+    /////////////////////////////////
+
+    // Broadcast via bluetooth the stored URL
+    private void broadcastUrl() {
+        Log.d(TAG, "broadcastUrl: " + shareUrl);
+        final AdvertiseData advertisementData = AdvertiseDataUtils.getAdvertisementData(shareUrl);
+        final AdvertiseSettings advertiseSettings = AdvertiseDataUtils.getAdvertiseSettings();
+        mBluetoothLeAdvertiser.stopAdvertising(mAdvertiseCallback);
+        mBluetoothLeAdvertiser.startAdvertising(advertiseSettings,
+            advertisementData, mAdvertiseCallback);
+    }
+
+    // Turn off URL broadcasting
+    public void disableUrlBroadcasting() {
+        Log.d(TAG, "disableUrlBroadcasting");
+        mBluetoothLeAdvertiser.stopAdvertising(mAdvertiseCallback);
+        mNotificationManager.cancel(NOTIFICATION_ID);
+    }
+
+
+
+    // Surface a notification to the user that a URL is being broadcast
+    // The notification specifies the URL being broadcast (the long URL)
+    // and cannot be swiped away
+    private void createNotification() {
+                //.setPriority(NotificationCompat.PRIORITY_MIN);
+        Intent resultIntent = new Intent();
+        TaskStackBuilder stackBuilder = TaskStackBuilder.create(this);
+        stackBuilder.addParentStack(BroadcastActivity.class);
+        stackBuilder.addNextIntent(resultIntent);
+        PendingIntent resultPendingIntent = stackBuilder.getPendingIntent(0,
+            PendingIntent.FLAG_UPDATE_CURRENT);
+        registerReceiver(stopServiceReceiver, new IntentFilter("myFilter"));
+        PendingIntent pIntent = PendingIntent.getBroadcast(this, 0, new Intent("myFilter"),
+            PendingIntent.FLAG_UPDATE_CURRENT);
+        NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this)
+                .setSmallIcon(R.drawable.ic_leak_add_white_24dp)
+                .setContentTitle(getString(R.string.broadcasting2))
+                .setContentText(mdisplayUrl)
+                .setOngoing(true)
+                .addAction(2, "Stop", pIntent);
```

Comment:
Use a string resource for "Stop" .  Also use R.id.<identifier> to get the id here.